detectors/node/opentelemetry-resource-detector-container/src/detectors/ContainerDetector.ts

```diff
@@ -42,30 +44,59 @@ export class DockerCGroupV1Detector implements Detector {
           });
     } catch (e) {
       diag.info(
-        'Docker CGROUP V1 Detector did not identify running inside a supported docker container, no docker attributes will be added to resource: ',
+        'Container Detector did not identify running inside a supported container, no container attributes will be added to resource: ',
         e
       );
       return Resource.empty();
     }
   }
 
+  private async _getContainerIdV1() {
+    const rawData = await ContainerDetector.readFileAsync(
+      this.DEFAULT_CGROUP_V1_PATH,
+      this.UTF8_UNICODE
+    );
+    const splitData = rawData.trim().split('\n');
+    for (const str of splitData) {
+      if (str.length >= this.CONTAINER_ID_LENGTH) {
+        return str.substring(str.length - this.CONTAINER_ID_LENGTH);
+      }
+    }
+    return undefined;
+  }
+
+  private async _getContainerIdV2() {
+    const rawData = await ContainerDetector.readFileAsync(
+      this.DEFAULT_CGROUP_V2_PATH,
+      this.UTF8_UNICODE
+    );
+    const str = rawData
+      .trim()
+      .split('\n')
+      .find(s => s.includes(this.HOSTNAME));
+    const containerIdStr = str
+      ?.split('/')
+      .find(s => s.length >= this.CONTAINER_ID_LENGTH);
```

Comment:
So we are looking for any string that has the correct length?
I think I've seen examples of `/proc/self/cgroup` that have truncated container ID or that have container id with a prefix(`docker-`) or suffix(`.slice`).

See some pseudo-examples(the length isn't checked in that implementation, and I don't know whether I would even, but you get the idea): https://github.com/signalfx/splunk-otel-js/blob/main/test/resource.test.ts

Reply:
I think contents for both `cgroup` and `mountinfo` files is a implementation detail of whatever calls the container runtime APIs.

Reply:
@Rauno56 Sorry - was out for a few days - so in terms of changes requested at this time - is to add more test cases around the files itself as in the examples right ?

Current implementation is check for keyword "hostname" and 64 length str - which may need to be updated depends on other containers.

Reply:
I actually totally missed the fact that you are filtering the rows to find the one with "hostname". It doesn't have to be perfect. Sorry for unproductive feedback.

What about `>=`? will it not be invalid id if the lenght = 67?

Reply:
@abhee11, your thoughts on my last comment?

Reply:
@Rauno56 Honestly - I dont have a good answer for this as we breaking down the string with "/" delimeter. 
We just return 64 length substring, it does seem a little crass but this is the best logic we could comeup with.

Reply:
I will keep an eye out in case we could make this logic more robust in the future, but current implementation in dockerv1 and how aws is doing is similar

Reply:
Sorry, I meant why we do we accept ids that are longer than 64 chars?

Reply:
@Rauno56 Valid, I think it is just the case we "consider everything >=64 " ( we do not expect to get anything more than 64 ) but if we do - we just return substr of that -- but i see your point and can change it to "==" check

Reply:
@Rauno56 I have made the "===" change as it made sense. Once it does pass the check please merge it if everything looks good.